lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1637,6 +1637,51 @@ class DecomposeAtenIndexPutOp : public OpRewritePattern<AtenIndexPutOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.index_add` op into `valsem.aten.index_put_impl` op. with
+// accumulate set to True
+class DecomposeAtenIndexAddOp : public OpRewritePattern<AtenIndexAddOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenIndexAdd op,
+                                PatternRewriter &rewriter) const override {
+
+    Value input = op.self();
+    Value updates = op.values();
+    int inputRank = getTensorRank(input);
+
+    if (inputRank < 0) {
+      return rewriter.notifyMatchFailure(op, "Invalid Input: Unranked tensor");
+    }
+
+    if (getTensorRank(updates) != inputRank) {
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented: Ranks of input and updates do not match");
+    }
+
+    SmallVector<int64_t> expandedUpdatesSizes{updates.getShape().front(), 1};
```

Comment:
Unused variable

Reply:
I'm getting `no member named 'getShape' in 'mlir::Value'` as the error here after redefining updates as `Value updates = op.source();`